airflow/www/views.py

```diff
@@ -1838,10 +1839,15 @@ def trigger(self, session=None):
                 form=form,
                 is_dag_run_conf_overrides_params=is_dag_run_conf_overrides_params,
             )
-
-        dr = DagRun.find(dag_id=dag_id, execution_date=execution_date, run_type=DagRunType.MANUAL)
+        # if run_id is not None, filter dag runs based on run id and ignore execution date
+        dr = DagRun.find(
+            dag_id=dag_id,
+            run_type=DagRunType.MANUAL,
+            run_id=run_id,
+            execution_date=execution_date if run_id is None else None,
```

Comment:
We still need to keep the `execution_date` clause when `run_id` is not None for now since two different run IDs with matching `execution_date` will fail.

Reply:
The reason I made this change is because the  DagRun.find (looking at existing runs and checking the run_id uniqueness) returns empty results when we send both run_id and execution_date since the combination is not found.

what I tried to do here is:
1. if there is a run_id - set execution_date to None and filter by the dag_id+run_id+run_type
2. if run_id is None - filter by dag_id+execution_date+run_type

Reply:
I think we need to ditch `DagRun.find()` and use direct SQLAlchemy lookup to do an `OR` on `run_id` and `execution_date` matches.

Reply:
can you point me to a code snippet using SQLAlchemy?